### PR TITLE
consul: cleanup and simplify leader election

### PR DESCRIPTION
**Pull request by adamdecaf**

### CODE REVIEW

File: internal/consul/leader.go [hunk 1]
```diff
@@ -23,17 +23,20 @@ import (
...
+		Session: c.session.ID,
 	}, nil)
 	if err != nil {
 		return err
 	}
 	if isLeader {
 		return nil
 	}
-	return logger.Info().LogErrorf("%s is not the leader", client.NodeId).Err()
+	return c.logger.Error().With(log.Fields{
```
Comment (darwinz) [inline internal/consul/leader.go:38]: Should this be an Error?  Won't hosts always try to check if they can be the leader and this should just inform it that there's already a leader?

Comment (adamdecaf) [reply internal/consul/leader.go:38]: Oh yea, forgot to set that back as Info.

Comment (darwinz) [reply internal/consul/leader.go:38]: Ignore my comment, I see that we're returning nil if the host can acquire a lock.  This makes sense to me actually.

Comment (adamdecaf) [reply internal/consul/leader.go]: You're right though. It's not an Error if we are unable to obtain leadership.

**Review by darwinz — APPROVED** [review]

**Pull request merged**